packages/core/core-flows/src/cart/steps/reserve-inventory.ts

```diff
@@ -80,7 +79,9 @@ export const reserveInventoryStep = createStep(
       }
     })
 
-    const reservations = await locking.execute(inventoryItemIds, async () => {
+    const lockingKeys = Array.from(new Set(inventoryItemIds))
+
+    const reservations = await locking.execute(lockingKeys, async () => {
```

Comment:
comment: if we create multiple reservations of the same inventory item in this step (one for each item in a cart that has items which share the inventory item) we need to deduplicate keys otherwise we will have deadlock. We need to update this in few more places if you agree with the approach.

Reply:
Nice catch.